x-pack/legacy/plugins/uptime/public/state/api/overview_filters.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+import { isRight } from 'fp-ts/lib/Either';
+import { GetOverviewFiltersPayload } from '../actions/overview_filters';
+import { getApiPath, parameterizeValues } from '../../lib/helper';
+import { OverviewFiltersType } from '../../../common/runtime_types';
+
+type ApiRequest = GetOverviewFiltersPayload & {
+  basePath: string;
+};
+
+export const fetchOverviewFilters = async ({
+  basePath,
+  dateRangeStart,
+  dateRangeEnd,
+  search,
+  schemes,
+  locations,
+  ports,
+  tags,
+}: ApiRequest) => {
+  const url = getApiPath(`/api/uptime/filters`, basePath);
+  const filterParams = parameterizeValues({
+    schemes,
+    locations,
+    ports,
+    tags,
+  });
+
+  const requiredParams = `?dateRangeStart=${dateRangeStart}&dateRangeEnd=${dateRangeEnd}`;
```

Comment:
Why are we encoding parameters using string concatenation at all and not a library that can take an Object or map and turn them into params? Can we use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
100% agree, this implementation is lazy; thanks for pointing this out, I can only think that in such a large diff I just didn't give enough thought to this. Will push a commit making the change and reply again.

Reply:
Check https://github.com/elastic/kibana/pull/50837/commits/22b34f4eea29363457e4cff69351ebae3bca5fa9 for resolution of this problem.